deltachat-rpc-client/src/deltachat_rpc_client/rpc.py

```diff
@@ -13,6 +13,48 @@ class JsonRpcError(Exception):
     pass
 
 
+class RpcFuture:
+    def __init__(self, rpc: "Rpc", request_id: int, event: Event):
+        self.rpc = rpc
+        self.request_id = request_id
+        self.event = event
+
+    def __call__(self):
+        self.event.wait()
+        response = self.rpc.request_results.pop(self.request_id)
+        if "error" in response:
+            raise JsonRpcError(response["error"])
+        if "result" in response:
+            return response["result"]
+        return None
+
+
+class RpcMethod:
+    def __init__(self, rpc: "Rpc", method: str):
+        self.rpc = rpc
+        self.method = method
```

Comment:
using "self.name" is better  -- this is already the  *Method* class and having a method attribute is weird then.